group/assignment6/RearrangeCars.cpp

```diff
@@ -35,36 +31,29 @@ void RearrangeCars::moveCar(vector<int> &currentStates, vector<int> &carPosition
 	carPosition[currentStates[to]] = to;
 }
 
-void RearrangeCars::generateSequenceOfMoves(const vector<int> &endStates, vector<Move> &sequenceOfMoves) {
-	vector<int> currentStates = this->startStates; // startStates will remain unchanged
-	vector<int> carPosition = findCarPositions(currentStates);
-
-	int emptySlotId = findEmptySlotId();
-	int currCarId, currCarSlotId;
-
-	while (true) {
-		currCarId = endStates[emptySlotId];
-		if (currCarId == -1)	// We reached slot which should be empty in the end
-			currCarId = findMisplacedCar(carPosition, endStates);
-		if (currCarId == -1) break;
-		currCarSlotId = carPosition[currCarId];
-		moveCar(currentStates, carPosition, currCarSlotId, emptySlotId);
-		sequenceOfMoves.push_back(Move(currCarId, make_pair(currCarSlotId, emptySlotId)));
-		emptySlotId = currCarSlotId;
-	}
-}
+bool RearrangeCars::validMove(int carId, int emptySlotId, const vector<set<int>>* constrains) {

```

Comment:
function name should be isMoveValid I think :) for me it's more understandable

Reply:
I agree here.